pkg/util/rpcdebug/interceptors_test.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2016-2023, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rpcdebug
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/grpc"
+
+	pulumirpc "github.com/pulumi/pulumi/sdk/v3/proto/go"
+)
+
+func TestClientInterceptorCatchesErrors(t *testing.T) {
+	tmp := t.TempDir()
+	logFile := filepath.Join(tmp, "test.log")
+	t.Cleanup(func() {
+		os.RemoveAll(tmp)
+	})
+
+	i, err := NewDebugInterceptor(DebugInterceptorOptions{
+		LogFile: logFile,
+	})
+
+	uci := i.DebugClientInterceptor(LogOptions{})
+
+	ctx := context.Background()
+
+	var inner grpc.UnaryInvoker = func(
+		ctx context.Context,
+		method string,
+		req, reply interface{},
+		cc *grpc.ClientConn,
+		opts ...grpc.CallOption,
+	) error {
+		return fmt.Errorf("ERR")
+	}
+
+	err = uci(ctx, "/pulumirpc.ResourceProvider/Configure",
+		&pulumirpc.ConfigureRequest{
+			Variables: map[string]string{"x": "y"},
+		},
+		&pulumirpc.ConfigureResponse{}, nil, inner)
+
+	assert.Error(t, err)
+
+	log, err := os.ReadFile(logFile)
+	assert.NoError(t, err)
```

Comment:
We should `require.NoError` here so that the test doesn't proceed if this fails.

```go
import "github.com/stretchr/testify/require"
```

```suggestion
	require.NoError(t, err)
```